### ISSUE TITLE
Adding the visible sun with an intensity divisor causing banding

### ISSUE THREAD

**Issue by jackhumbert** [BastiaanOlij/godot-sky-asset#7]
Hey Bastiaan! Love your work on this and Godot tutorials in general. I've been working on a couple lines in the fragment to add the visible sun:

	// divide by the intensity of the background to get accurate colors
	color /= intensity;
	
	// Add the visible sun
	float sun_aa = 0.0093084226799999992413;
	float mu = max(sun_aa - acos(dot(dir, sun_pos) / (length(dir) * length(sun_pos))), 0.0);
	color = color + (mu / sun_aa * 2.0 * vec3(1, 1, 1));

`intensity` is set from the script, and matches the `background_energy` of the environment, so that I can get a neat glare effect [using something like this](https://github.com/SIsilicon/Godot-Lens-Flare-Plugin). 

The problem is that when I set the intensity higher to try to get an accurate brightness ratio, I get a lot of banding in the background due to the limited bit depth of the image. I was curious if you could see any obvious workarounds to an approach like this. I think traditionally a lens flare would be added in postprocessing with the known sun position, but this seemed like it might be a quick way to do this, since I'm already doing a lens flare for other bright things in the game.

**Comment by jackhumbert** [comment]
I figured out one way to do this that I'm pretty happy with - I moved the code to a parent node and duplicated the viewport. I have it adjusting the first for the intensity of the background (so the sun is super bright), which is used for the ambient light in the scene. Then I have a custom fog shader that pulls in the image from the second viewport to fade into the custom sky as things are farther away. I'll put a link up here when I have something ready to share!
